agent/provider/src/cli/rule.rs

```diff
@@ -40,16 +40,33 @@ pub enum SetOutboundRule {
         #[structopt(short, long, possible_values = Mode::VARIANTS)]
         mode: Mode,
     },
-    Partner(RuleWithCert),
+    Partner(PartnerRuleWithCert),
 }
 
 #[derive(StructOpt, Clone, Debug)]
-pub enum RuleWithCert {
-    CertId {
-        cert_id: String,
+pub struct CertId {
+    cert_id: String,
+    #[structopt(short, long, possible_values = Mode::VARIANTS)]
+    mode: Mode,
+}
+
+#[derive(StructOpt, Clone, Debug)]
+pub enum AuditedPayloadRuleWithCert {
+    CertId(CertId),
+    ImportCert {
+        import_cert: PathBuf,
         #[structopt(short, long, possible_values = Mode::VARIANTS)]
         mode: Mode,
+        /// When importing chain of X.509 certificates set rule to every certificate in a chain.
+        /// By default rule is assigned only to last (leaf) certificate of a chain.
+        #[structopt(short, long)]
+        whole_chain: bool,
     },
+}
+
+#[derive(StructOpt, Clone, Debug)]
+pub enum PartnerRuleWithCert {
```

Comment:
`RuleWithCert` enum now has two versions, because in case of `audited-payload` there will be `--whole-chain` param.